### DISCUSSION TITLE
Remote reference and custom rules?

### DISCUSSION THREAD

**Discussion by diskun00** [Tufin/oasdiff#633] [Q&A]
Hi, I'm wondering if oasdiff support remote reference components with auth. I checked the document but didn't find anything related.

besides, if we want to add custom check rules, we would need to build the oasdiff with the new customecheck according to this [doc](https://github.com/Tufin/oasdiff/blob/main/docs/CUSTOMIZING-CHECKS.md), right? that means, one either build the oasdiff themselves or get their changes merged to repo to apply new custom rule?

**Comment by reuvenharrison** [comment]
Remote refs:
I am not sure.
Can you provide an example?


Custom checks:
Yes, the best way is to submit an issue and/or PR with the required checks.

**Reply by diskun00** [reply]
Thank you @reuvenharrison for the quick answer! One example of remote ref is
```yml
components:
  schemas:
    AwesomeClickEvent:
      type: object
      properties:
        awesome_id:
          type: integer
          description: The ID of the event that was clicked.
        filter:
          $ref: '#/components/schemas/SearchFilter'
    SearchFilter:
      $ref: 'http://localhost:8000/diff/1_search_result_v1.yml#/components/schemas/SearchResult'

```

in this case if we need auth for server http://localhost:8000/, could oasdiff use credential to access such resources?

**Comment by reuvenharrison** [comment]
I think this should be handled by the underlying https://github.com/getkin/kin-openapi library which we use to load and parse openapi specs.
It doesn't seem to be supported currently/
